merlin-sdk/src/test/java/gov/nasa/jpl/aerie/merlin/protocol/types/SerializedValueTest.java

```diff
@@ -0,0 +1,26 @@
+package gov.nasa.jpl.aerie.merlin.protocol.types;
+
+import java.util.List;
+import java.util.Map;
+
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public final class SerializedValueTest {
+  @Test
+  public void testHashCode() {
+    final var someValue1 = SerializedValue.NULL;
+    final var someValue2 = SerializedValue.NULL;
+    final var list1 = List.of(someValue1, SerializedValue.of(42));
+    final var list2 = List.of(someValue2, SerializedValue.of(43));
+    final var map1 = Map.of(someValue1, SerializedValue.of(42));
+    final var map2 = Map.of(someValue2, SerializedValue.of(43));
+    final var map3 = Map.of(someValue2, SerializedValue.of(42));
+
+    assertThat(someValue1.hashCode()).isEqualTo(someValue2.hashCode());
+    assertThat(list1.hashCode()).isNotEqualTo(list2.hashCode());
+    assertThat(map1.hashCode()).isNotEqualTo(map2.hashCode());
```

Comment:
It is legal (and does not harm correctness) for these hash codes to be the same. The relevant contract is "if `list1.equals(list2)` then `list1.hashCode() == list2.hashCode()`. If the two objects are not `equals()`, then their hash codes can be anything.